app/Services/PathStationService.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Services;
+
+use App\Path;
+
+class PathStationService
+{
+    /**
+     * Attaches array of Station ids and relative travel time to each to provided Path model.
+     */
+    public function attachStations(array $stations, Path $path): void
+    {
+        foreach ($stations as $station) {
+            $path->stations()->syncWithoutDetaching([$station['id'] => ['travel_time' => $station['travel_time']]]);
+        }
+    }
+
+    /**
+     * Detaches array of Stations from provided Path model by provided list of station ids.
+     */
+    public function detachStations(array $stations, Path $path): void
+    {
+        foreach ($stations as $station) {
+            $path->stations()->detach($station['id']);
+        }
+    }
+
+    /**
+     * Shows Stations attached to a provided Path model and appends relative travel_time.
+     */
+    public function showAttachedStations(Path $path): array
+    {
+        $stations = $path->stations->toArray();
+
+        foreach ($stations as $key => $val) {
+            $stations[$key]['travel_time'] = $path->stations()->wherePivot('station_id', '=', $val['id'])->first()->pivot->travel_time;
```

Comment:
How many queries to database are you calling with that? Protip: if you have no idea, installing Telescope would be a good thing.